2-higher-order-functions/eloquent/exercise.js

```diff
@@ -1,20 +1,40 @@
 // Challenge 1. Flattening
 let arrays = [[1, 2, 3], [4, 5], [6]];
 
-// Your code here.
+arrays.reduce((a, c) => a.concat(c), []);
 // → [1, 2, 3, 4, 5, 6]
 
 // Challenge 2. Your own loop
 // Your code here.
 
+function loop(n, condition, change, fn) {
+  while (condition(n)) {
+    fn(n);
+    n = change(n);
+  }
+}
+
 loop(3, n => n > 0, n => n - 1, console.log);
 // → 3
 // → 2
 // → 1
 
 // Challenge 3. Everything
 function every(array, test) {
-  // Your code here.
+  let arr = [];
+  for (element of array) {
+      if (test(element)) arr.push(element);
+  }
+  return arr.length == array.length;
+}
+
+function every(array, test) {
+  let arr = [];
+  let elementsArray = [...array].map(e => [e]);
+  elementsArray.forEach(e => {
+    if (e.some(a => test(a))) arr.push(e[0]);
+  });
+  return arr.length == array.length;
 }
```

Comment:
- Why clone an array (you are using `map` that return new array)

Reply:
Re-write `every` second implementation.

Reply:
Changes some names and spacing. One request please check the logic once. Because it's kind of "Jugaru" type of code 😂 that's why I need a suggestion for this one.